libnmstate/state.py

```diff
@@ -155,6 +225,76 @@ def merge_interfaces(self, other_state):
             dict_update(other_state.interfaces[name], self.interfaces[name])
             self._ifaces_state[name] = other_state.interfaces[name]
 
+    def merge_route_config(self, other):
+        """
+        Merge routes from other to self.
+        If interface does exists in self, create an empty one.
+        Delete route entries if 'state: absent'.
+        If certain property missing when deleting, treat it as wildcard.
+        """
+        origin_other = copy.deepcopy(other)
+        # Copy other to self if certain interface does not defined.
+        for iface_name, routes in six.viewitems(other.iface_routes):
+            if iface_name not in self.iface_routes:
+                self.routes.extend(routes)
+
+        # Handle 'state: absent' without next-hop interface defined
+        for route in copy.deepcopy(self.routes):
+            if route.get(Route.NEXT_HOP_INTERFACE):
```

Comment:
IMHO a combined if statement with `or` would be better to read.

Reply:
I intend to reduce the indent level as linux kernel(using hard tab) suggested.
But anyway, you will get it.